tests/drivers/can/api/boards/numaker_m2l31ki.conf

```diff
@@ -0,0 +1 @@
+CONFIG_TEST_USERSPACE=n
```

Comment:
This should be handled on the board level. Otherwise we will need to have this line in many, many tests across the tree.

Reply:
Similar example of a board/MCU having too few MPU regions, handled in the board definition: https://github.com/zephyrproject-rtos/zephyr/blob/7abb9d7593b7f4d727fa1fa6440a6d87f612f99a/boards/nxp/twr_ke18f/Kconfig.defconfig#L14

Reply:
@henrikbrixandersen ,
thanks of your advice.
Even though I tried to set `CONFIG_TEST_USERSPACE=n` in `zephyr/boards/nuvoton/numaker_m2l31ki` or in `zephyr/soc/nuvoton/numaker/m2l31x`, it's still override by `zephyr/tests/drivers/can/api/prj.conf`.
So, except `tests/drivers/can/api/boards/numaker_m2l31ki.conf`, how could  override prj.conf ?

Reply:
@henrikbrixandersen ,
if set CONFIG_USERSPACE=n in zephyr/boards/nuvoton/numaker_m2l31ki, CONFIG_TEST_USERSPACE will not be effective.
However, loose the benefit of CONFIG_USERSPACE=y.
How about your opinion ?

Reply:
@henrikbrixandersen ,
it's done to set the configuration on board level instead of on each test.